modules/check_in/spec/request/v2/pre_check_ins_request_spec.rb

```diff
@@ -161,4 +153,48 @@
       end
     end
   end
+
+  describe 'POST `create`' do
+    let(:post_params) do
+      {
+        pre_check_in: {
+          uuid: id,
+          demographics_up_to_date: true,
+          next_of_kin_up_to_date: true,
+          check_in_type: :pre_check_in
+        }
+      }
+    end
+
+    context 'when session is authorized' do
+      let(:success_resp) { Faraday::Response.new(body: 'Pre-checkin successful', status: 200) }
+
+      before do
+        allow_any_instance_of(CheckIn::V2::Session).to receive(:authorized?).and_return(true)
+        allow_any_instance_of(::V2::Chip::Service).to receive(:pre_check_in).and_return(success_resp)
+      end
+
+      it 'returns successful response' do
+        post '/check_in/v2/pre_check_ins', params: post_params
+
+        expect(response.body).to eq(success_resp.to_json)
+      end
+    end
+
+    context 'when session is not authorized' do
+      let(:body) { { 'permissions' => 'read.none', 'status' => 'success', 'uuid' => id } }
+      let(:unauth_response) { Faraday::Response.new(body: body, status: 401) }
+
+      before do
+        allow_any_instance_of(::V2::Chip::Service).to receive(:pre_check_in).and_return(unauth_response)
+      end
+
+      it 'returns unauthorized response' do
+        post '/check_in/v2/pre_check_ins', params: post_params
+
+        expect(response.body).to eq(unauth_response.body.to_json)
+        expect(response.status).to eq(401)
```

Comment:
nit: may be return unauth_response.status instead of 401 directly

Reply:
updated in 2f84e5f50a810f003c289662a79e79a1d922b0ed